Check most recent build on default branch

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the Python code to check the most recent build _on the default branch_ when determining build status.

## 💭 Motivation and context ##

Resolves #68.

## 🧪 Testing ##

All automated tests pass.  I also ran the code locally (commenting out the actual call to kick off a new APB build) and verified that it functions as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release (necessary if and only if the version was bumped).

jsf9k committed:
Check only most recent run on default branch

When the Markdown output is generated, the code previously determined
the failure status by checking if the most recent build passed or
failed. Often this build was a Dependabot or other PR build whose
failure should not determine the build status of the entire build. It
is better to instead check the last build that was built from the
default branch.

Now the Markdown output shows a failure for a project if and only if
the project is genuinely broken, i.e., if the default branch fails to
build.

See #68 for more details.

jsf9k committed:
Bump version from 2.0.1 to 2.1.0

mcdonnnj commented on src/apb/entrypoint.py:
```diff
@@ -51,10 +60,13 @@ def get_last_run(
         )
         return None
     workflow_runs = response.json()["workflow_runs"]
```
I think this should be folded into the assignment of `workflow_runs_on_default_branch` as we don't do anything else with the variable.

jsf9k replied:
Please see commit cf6d9f6.

mcdonnnj commented on src/apb/entrypoint.py:
```diff
@@ -51,10 +60,13 @@ def get_last_run(
         return None
     workflow_runs = response.json()["workflow_runs"]
-    if len(workflow_runs) == 0:
+    workflow_runs_on_default_branch = [
```
At this point would it be better to break out of a while loop once the first run on the default branch is found instead of needlessly processing all of the runs? Predefine the variable as `None` and turn the length check into an `is not None` check and I think it's more efficient since we only care about the first run we find.

jsf9k replied:
Please see commit 7c93b04.

mcdonnnj reviewed (changes requested):
Thanks for tackling this problem. I have a couple of thoughts (that are technically mutually exclusive) on how this is implemented in the `get_last_run_on_default_branch()` function for your consideration.

jsf9k committed:
Remove unused variable

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Replace list comprehension with a while loop

We can break out of the while loop once we find a valid workflow run,
thus avoiding the processing of further list elements.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

dav3r reviewed (approved):
👍

mcdonnnj commented on src/apb/entrypoint.py:
```diff
@@ -50,12 +59,15 @@ def get_last_run(
-        last_run_date = workflow_runs[0]["created_at"]
-        return isoparse(last_run_date).replace(tzinfo=None)
+
+    # Find the date of the most recent workflow run against the default branch.
```
Nitpick but we typically omit periods for one-line comments.

```suggestion
    # Find the date of the most recent workflow run against the default branch
```

mcdonnnj reviewed (approved):
LGTM ✔

jsf9k committed:
Remove period from single-line comment

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>
